### ISSUE TITLE
Load JS from footer or deferred (when using no bundling)

### ISSUE THREAD

**Issue by stefandoorn** [algolia/algoliasearch-magento-2#784]
I currently see 3 JS files being loaded from the `<head>` without using any deferring techniques. We have bundling disabled for testing purposes. I think from a performance perspective, it would be better to push this loading either deferred or from the end of the HTML body. 

Using the latest version of Magento 2.3.x and the latest Algolia M2 release.

```
<script  type="text/javascript"  src="https://domain/static/version1559042027/frontend/xxx/xxx/nl_NL/Algolia_AlgoliaSearch/internals/common.min.js"></script>
<script  type="text/javascript"  src="https://domain/static/version1559042027/frontend/xxx/xxx/nl_NL/Algolia_AlgoliaSearch/instantsearch.min.js"></script>
<script  type="text/javascript"  src="https://domain/static/version1559042027/frontend/xxx/xxx/nl_NL/Algolia_AlgoliaSearch/autocomplete.min.js"></script>
```

Am I overlooking some settings here?

**Comment by brightonmike** [comment]
@stefandoorn 

I'm overriding the XML and adding defer. This should really be in the module however so a +1.

```
        <script defer="defer" src="Algolia_AlgoliaSearch::internals/common.js"/>
        <script defer="defer" src="Algolia_AlgoliaSearch::instantsearch.js"/>
        <script defer="defer" src="Algolia_AlgoliaSearch::autocomplete.js"/>
```

**Comment by damcou** [comment]
Hello @stefandoorn and @brightonmike ,

Thanks a lot for your investigations regarding the inclusion of the javascript files.
We are currently working on a frontend refactoring of the extension and we will take your advices into consideration. Unfortunately, I can't give any ETA for now. We'll keep you updated.

Thanks for understanding.
Have a good day.
Best.

**Comment by stefandoorn** [comment]
Hi @damcou, thanks. Any updates on this?

**Comment by damcou** [comment]
Hello @stefandoorn ,
We are still working on it with our javascript experts, but unfortunately, I can't guarantee any ETA for the moment.

Sorry for that.
Best.

**Comment by pipe-devnull** [comment]
Hi, 

Any chance you can include all JS using requireJs? I'm using advanced JS bundling and I can't currently include Algolia JS in those bundles because of the way you are including the files.

I could override and change this but I'd rather not risk it breaking in future updates etc.

Thanks,

**Comment by damcou** [comment]
Hi there,

We are still working on the inclusion of the JS files and the changes should be part of our next big release (2.0.0).
As this request seems to be the same as this one https://github.com/algolia/algoliasearch-magento-2/issues/642 which has already been tagged to be released in 2.0.0, I close this one, feel free to follow-up there.

Best.
